cron/format/schema_gen.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package format
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/mcuadros/go-jsonschema-generator"
+)
+
+// https://github.com/googleapis/google-cloud-go/blob/bigquery/v1.30.0/bigquery/schema.go#L544
+type bigQueryJSONField struct {
+	Description string              `json:"description"`
+	Mode        string              `json:"mode"`
+	Name        string              `json:"name"`
+	Type        string              `json:"type"`
+	Fields      []bigQueryJSONField `json:"fields,omitempty"`
+}
+
+func generateSchema(schema bigquery.Schema) []bigQueryJSONField {
```

Comment:
I find it weird that we need to implement our custom logic for generating BQ schema from `bigquery.Schema` struct. We should ideally not need to maintain this code since the internals of BQ schema implementation could change over time. There needs to be an exported fn to do this.

Could you file an issue with https://github.com/googleapis/google-cloud-go for this? Or maybe leave a TODO in our code to look into this.

Reply:
agreed. issue created https://github.com/googleapis/google-cloud-go/issues/5833